src/mscorlib/src/System/IO/__Error.cs

```diff
@@ -120,24 +121,27 @@ internal static void WinIOError(int errorCode, String str)
                     throw new DriveNotFoundException(SR.Format(SR.IO_DriveNotFound_Drive, str));
 
                 case Win32Native.ERROR_INVALID_PARAMETER:
-                    throw new IOException(Win32Native.GetMessage(errorCode), Win32Native.MakeHRFromErrorCode(errorCode), str);
+                    Debug.Assert(String.IsNullOrEmpty(str), "message should include path");
```

Comment:
Do you plan to keep these asserts?

Note that there is almost identical code in GetExceptionForWin32Error in the shared partition that does not have asserts like this.